metrics.py updates

DavidKelly999 opened the pull request:
Fix some missing return statements and misnamed function calls
Dictionary values() are not json serializable, so convert them to a list
In c2m2_assessment, some 'value' values are None because the assessment was skipped - set those to optional.

karlcz commented on cfde_deriva/metrics.py:
```diff
@@ -117,7 +129,7 @@ def register_datapackage_measurements(registry_catalog, submission_id, records):
     for record in records:
         metric = _get_required(record, 'metric')
-        value = _get_required_number(record, 'value')
+        value = _get_optional_number(record, 'value')
```
I'm not sure why we would register a metric that has not been evaluted... shouldn't we just skip those?

karlcz commented on cfde_deriva/metrics.py:
```diff
@@ -156,17 +168,11 @@ def register_datapackage_measurements(registry_catalog, submission_id, records):
         json=need_update,
-    ).json() # discard response data
-
-def get_datapackage_measurements(registry_catalog, submission_id, headers=DEFAULT_HEADERS):
```
Please rebase your change on master so this additional headers param is not pruned out in the PR...

DavidKelly999 commented:
I've made some changes to my script. Dictionaries with 'value' set to None won't be added. I've switched back to 'value' being required and added the missing header info.
